networktest.go

```diff
@@ -1270,6 +1270,45 @@ func (n *networkHarness) CloseChannel(ctx context.Context,
 	lnNode *lightningNode, cp *lnrpc.ChannelPoint,
 	force bool) (lnrpc.Lightning_CloseChannelClient, *chainhash.Hash, error) {
 
+	// Create a channel outpoint that we can use to compare to channels
+	// from the ListChannelsResponse.
+	fundingTxID, err := chainhash.NewHash(cp.FundingTxid)
+	if err != nil {
+		return nil, nil, err
+	}
+	chanPoint := wire.OutPoint{
+		Hash:  *fundingTxID,
+		Index: cp.OutputIndex,
+	}
+
+	// If we are not force closing the channel, wait for channel to become
+	// active before attempting to close it.
+	numTries := 10
+CheckActive:
+	for i := 0; !force && i < numTries; i++ {
+		listReq := &lnrpc.ListChannelsRequest{}
+		listResp, err := lnNode.ListChannels(ctx, listReq)
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable fetch node's "+
+				"channels: %v", err)
+		}
+
+		for _, c := range listResp.Channels {
+			if c.ChannelPoint == chanPoint.String() && c.Active {
+				break CheckActive
```

Comment:
This label isn't necessary, there's no outer loop involved. As a result, we can just use a regular break here.

Reply:
There are actually two loops, the one looping over channels, and the one iterating 10 retries. We want to stop doing retries in this case.

Reply:
Ahh I see now, my initial diff view didn't show the entire func!